httpx/config.py

```diff
@@ -223,6 +236,43 @@ def __repr__(self) -> str:
         )
 
 
+class HTTPVersionConfig:
+    """
+    Configure which HTTP protocol versions are supported.
+    """
+
+    def __init__(self, http_versions: HTTPVersionTypes = None):
+        if http_versions is None:
+            http_versions = ["HTTP/1.1", "HTTP/2"]
+
+        if isinstance(http_versions, str):
+            self.http_versions = {http_versions.upper()}
+        elif isinstance(http_versions, HTTPVersionConfig):
+            self.http_versions = http_versions.http_versions
+        else:
+            self.http_versions = {version.upper() for version in http_versions}
+
+        for version in self.http_versions:
+            if version not in ("HTTP/1.1", "HTTP/2"):
+                raise ValueError(f"Unsupported HTTP version {version!r}.")
+
+        if not self.http_versions:
+            raise ValueError(f"HTTP versions cannot be an empty list.")
+
+    @property
+    def alpn_strings(self) -> typing.List[str]:
```

Comment:
Thoughts on naming this `alpn_identifiers`?

Reply:
Yup. I'll sort it. 👍